src/types/protocol.rs

```diff
@@ -28,20 +28,30 @@ use std::{
 
 /// Protocol name.
 #[derive(Debug, Clone)]
+#[cfg_attr(feature = "fuzz", derive(serde::Serialize, serde::Deserialize))]
 pub enum ProtocolName {
+    #[cfg(not(feature = "fuzz"))]
```

Comment:
This was added because the proto name generated by the fuzzer could not be static?

Reply:
Serde cannot de-serialize ``&'static str`` and since we use serde to de-serialize fuzzer input, the ``Static`` variant cannot be supported.